src/main/java/org/broadinstitute/hellbender/tools/funcotator/FuncotatorDataSourceRetriever.java

```diff
@@ -0,0 +1,221 @@
+
+    @Argument(fullName = OVERWRITE_ARG_LONG_NAME,
+            shortName = OVERWRITE_ARG_LONG_NAME,
+            doc = "Overwrite output file if it exists already.", optional = true)
+    private boolean overwriteOutputFile = false;
+
+    //==================================================================================================================
+    // Constructors:
+
+    //==================================================================================================================
+    // Override Methods:
+
+    @Override
+    protected Object doWork() {
+
+        // Get the correct data source:
+        if ( getSomaticDataSources ) {
+            logger.info("Somatic data sources selected.");
+
+            // Get the germline datasources file:
+            final NioFileCopier xerox = downloadDatasources(SOMATIC_GCLOUD_DATASOURCES_PATH);
+
+            // Validate the data sources if requested:
+            if (validateIntegrity) {
+                logger.info("Integrity validation selected.");
+                validateIntegrity(xerox, SOMATIC_GCLOUD_DATASOURCES_SHA256_PATH);
+            }
+        }
+
+        if ( getGermlineDataSources ) {
+
+            logger.info("Germline data sources selected.");
+
+            // Get the germline datasources file:
+            final NioFileCopier xerox = downloadDatasources(GERMLINE_GCLOUD_DATASOURCES_PATH);
+
+            // Validate the data sources if requested:
+            if (validateIntegrity) {
+                logger.info("Integrity validation selected.");
+                validateIntegrity(xerox, GERMLINE_GCLOUD_DATASOURCES_SHA256_PATH);
+            }
+        }
+
+        logger.info("IMPORTANT: You must unzip the downloaded data sources prior to using them with Funcotator.");
+
+        // Token return value:
+        return true;
+    }
+
+    @Override
+    protected void onStartup() {
+        // Make sure the user specified at least one data source to download:
+        if ((!getSomaticDataSources) && (!getGermlineDataSources)) {
+            throw new UserException("Must select either somatic or germline datasources.");
+        }
+
+        if ( overwriteOutputFile ) {
+            logger.info("Overwrite ENABLED.  Will overwrite existing data sources download.");
+        }
+    }
+
+    //==================================================================================================================
+    // Static Methods:
+
+    //==================================================================================================================
+    // Instance Methods:
+
+    private NioFileCopier downloadDatasources(final Path dataSourcesPath) {
+
+        // Get the data sources file:
+        final Path outputDestination = getDataSourceLocalPath(dataSourcesPath);
+
+        // Set up and initiate our download:
+        final NioFileCopier xerox = NioFileCopier.create(dataSourcesPath, outputDestination, overwriteOutputFile);
```

Comment:
If the destination already exists and `overwriteOutputFile == false`, shouldn't the tool exit immediately?

Reply:
OBE due to reading the file directly.